rj_gameplay/stp/play/__init__.py

```diff
@@ -90,7 +90,7 @@ def assign_roles(
                 min_cost = 1e9
                 cheapest_robot = None
                 for robot in world_state.our_robots:
-                    if robot.id in used_robot_ids:
+                    if robot.__hash__() in used_robots:
```

Comment:
You should actually be able to just say this. Python will automatically call the hash function you wrote. (Typically all functions that start with __ are not meant to be called directly by the programmer in Python.)
```suggestion
                    if robot in used_robots:
```